src/router/index.js

```diff
@@ -9,6 +12,41 @@ const routes = [
     path: "/",
     name: "Home",
     component: Home,
+    children: [
+      {
+        path: "list/:listId",
+        name: "ListDetails",
+        component: ListDetails,
+        props: (route) => {
+          const allTasksByListId = store.tasksData.filter(
+            (task) => task.id_list === route.params.listId
+          );
+          return {
+            listName: store.listsData.find(
+              (list) => list.id === route.params.listId
+            ).name,
+            allTasks: [...allTasksByListId],
+          };
+        },
+        beforeEnter: (to, from, next) => {
+          // fetching list data goes here
+          const listsData = store.getAllLists();
+          // check if list id exist in lists data
+          const exists = listsData.find((list) => list.id === to.params.listId);
+          if (exists) {
+            return next();
+          } else {
+            return next({ name: "PageNotFound" });
+          }
```

Comment:
Optional: ternary operator can be used as a single expression.
```
return exists ? next() : next({ name: "PageNotFound" });
```

Reply:
Updated!